sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -179,7 +189,472 @@ func (suite *NetworkflowManagerTestSuite) TestAddNoOriginator() {
 	h.connectionsSequenceID = sequenceID
 
 	err := h.Process(networkInfoOpen, nowTimestamp, sequenceID)
-	suite.NoError(err)
+	s.NoError(err)
+
+	s.Len(h.endpoints, 1)
+}
+
+// endregion
+
+// region networkFlowManager tests
+
+func (s *NetworkFlowManagerTestSuite) TestEnrichConnection() {
+	mockCtrl := gomock.NewController(s.T())
+	m, mockEntityStore, mockExternalSrc, _ := createManager(mockCtrl)
+	srcID := "src-id"
+	dstID := "dst-id"
+	cases := map[string]struct {
+		connPair                    *connectionPair
+		enrichedConnections         map[networkConnIndicator]timestamp.MicroTS
+		expectEntityLookupContainer expectFn
+		expectEntityLookupEndpoint  expectFn
+		expectExternalLookup        expectFn
+		expectedIndicator           *networkConnIndicator
+		expectedConnection          *connection
+		expectedStatus              *connStatus
+	}{
+		"Rotten connection should return rotten status": {
+			connPair: createConnectionPair().incoming().external().firstSeen(timestamp.Now().Add(-maxContainerResolutionWaitPeriod * 2)),
+			expectEntityLookupContainer: func() {
+				mockEntityStore.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
+					return clusterentities.ContainerMetadata{}, false
+				})
+			},
+			expectedStatus: &connStatus{
+				rotten: true,
+			},
+		},
+		"Incoming external connection with unsuccessful lookup should return internet entity": {
+			connPair:            createConnectionPair().incoming().external(),
+			enrichedConnections: make(map[networkConnIndicator]timestamp.MicroTS),
+			expectEntityLookupContainer: func() {
+				mockEntityStore.EXPECT().LookupByContainerID(gomock.Any()).Times(1).DoAndReturn(func(_ any) (clusterentities.ContainerMetadata, bool) {
+					return clusterentities.ContainerMetadata{
+						DeploymentID: dstID,
+					}, true
+				})
+			},
+			expectExternalLookup: func() {
```

Comment:
WDYT about creating some setup functions for these calls? Something like
```go

func givenNetworkLookup(info *storage.NetworkEntityInfo) {
       mockExternalSrc.EXPECT().LookupByNetwork(gomock.Any()).Times(1).DoAndReturn(func(_ any) *storage.NetworkEntityInfo {
              return info
       })
}

// in the test case definition
expectExternalLookup: func() { givenNetworkLookup( /* define here */) }
// in another
expectExternalLookup: func() { givenNetworkLookup(nil) }
```

Just to remove a bit of the boilerplate here which makes reading the test a bit harder. You could return a function also to pass it directly
```go
expectExternalLookup: givenNetworkLookup(nil)
```

But that might conflict with the suggestion above of making a single `expect` block.